session-6/task5/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+func main() {
+
+	counts := countWords("go", "java", "go", "python", "go", "java")
+	fmt.Println("Output", counts)
+	for key, value := range counts {
+		fmt.Printf("%s : %d\n", key, value)
+	}
+}
+
+func countWords(words ...string) map[string]int {
+	fmt.Println("Input:", words)
+	counts := make(map[string]int)
+	receivedStrings := strings.Join(words, ", ")
+	for _, subString := range words {
+		matchedWord := strings.Count(receivedStrings, subString)
+		counts[subString] = matchedWord
+	}
+	return counts
+}
```

Comment:
Duzdur, netice eynidi. Amma burda taskin telebleri pozulub. 

Write a program that counts the frequency of each word in a given slice of strings.

Input: []string{"go", "java", "go", "python", "go", "java"}
Output: map[string]int{"go": 3, "java": 2, "python": 1}
Once the map is built, iterate through it and print each word with its count.

```suggestion
func main() {
	var langs = []string{"go", "java", "go", "python", "go", "java"}
	var result = map[string]int{}

	for _, value := range langs {
		result[value] += 1
	}

	fmt.Println(result)
}
```

Reply:
Bəli, inputda slice istifadə etməmişəm və sizin kod çox daha qısadır, pratiktir.